Makefile

```
@@ -307,3 +345,72 @@ verify-bundle: bundle ## Verify bundle update.
 .PHONY: verify-fmt
 verify-fmt: fmt ## Verify fmt update.
 	git diff --exit-code ./api ./controllers
+
+## local configurations
+
+deploy-develmode: manifests kustomize ## Deploy controller in debug mode to the K8s cluster specified in ~/.kube/config.
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${OPERATOR_IMAGE}
+	@if [ $(NAMESPACE) != '' ];then \
+		echo "Setting Custom Namespace: $(NAMESPACE)"; \
+		cd $(PROJECT_DIR)/config/default && $(KUSTOMIZE) edit set namespace $(NAMESPACE); \
+	fi
+
+	cd $(PROJECT_DIR) && $(KUSTOMIZE) build config/deploy > $(DEPLOYMENT_FILE)
+
+	$(KUSTOMIZE) build config/default | kubectl apply -f -
+
+# clean up
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${DEFAULT_OPERATOR_IMAGE}
+	@if [ $(NAMESPACE) != '' ];then \
+		echo "Removing Custom Namespace: $(NAMESPACE)"; \
+		cd $(PROJECT_DIR)/config/default && $(KUSTOMIZE) edit set namespace $(DEFAULT_REPO); \
+	fi
+
+.PHONY: local-cleanup
+local-cleanup:
+	$(MAKE) kind-delete-cluster
+
+.PHONY: local-env-setup
+local-env-setup:
+	$(MAKE) kind-delete-cluster
+	$(MAKE) kind-create-cluster
+	$(KUSTOMIZE) build config/crd | kubectl apply -f -
+
+.PHONY: local-setup
+local-setup: export OPERATOR_IMAGE := authorino-operator:dev
+local-setup:
+	$(MAKE) local-env-setup
+	$(MAKE) docker-build
+	echo "Deploying Authorino control plane"
+	$(KIND) load docker-image ${OPERATOR_IMAGE} --name ${KIND_CLUSTER_NAME}
+	$(MAKE) install
+	$(MAKE) deploy
+
+.PHONY: local-redeploy
+local-redeploy: export OPERATOR_IMAGE := authorino-operator:dev
+local-redeploy:
```

For consistency with how Authorino [calls](https://github.com/Kuadrant/authorino/blob/374cd7c42f80babc84e9f960be2bd9d8e63fde41/Makefile#L261) it:

```suggestion
local-rollout:
```